lib/ret/project.ex

```diff
@@ -116,8 +116,8 @@ defmodule Ret.Project do
       ) do
     project
     |> changeset(account, params)
-    |> put_change(:project_owned_file_id, project_owned_file.owned_file_id)
-    |> put_change(:thumbnail_owned_file_id, thumbnail_owned_file.owned_file_id)
+    |> put_assoc(:project_owned_file_id, project_owned_file)
+    |> put_assoc(:thumbnail_owned_file_id, thumbnail_owned_file)
```

Comment:
There are a couple of problems with this change. First, put_assoc will currently fail the second time you call it on a project, since these associations use the default `:raise` setting on the `:on_replace` option. This is currently why the CI build is failing on this PR.
Aside, from that, if you do want to go ahead with this change and set a different `:on_replace` option, you'll need to use `:project_owned_file` here instead of `:project_owned_file_id`.
See https://hexdocs.pm/ecto/Ecto.Changeset.html#module-the-on_replace-option and https://hexdocs.pm/ecto/Ecto.Schema.html#belongs_to/3 (although it's weird that we've stated a project `belongs_to` project_owned_file in the project schema)
I'd be inclined to keep using `put_change`, since it already does what we want.

Reply:
Ah good catch. Hadn't looked into the failing test yet (its usually a linting thing I just handle at the end). Since I ended up not actually needing to create projects this change was unneeded anyway, so I will just revert.